src/core/annotation.js

```diff
@@ -134,10 +136,12 @@ class AnnotationFactory {
       subtype,
       id,
       pdfManager,
-      acroForm: acroForm instanceof Dict ? acroForm : Dict.empty,
+      acroForm: acroFormDict,
       attachments,
       xfaDatasets,
       collectFields,
+      needAppearances:
+        !collectFields && (acroFormDict.get("NeedAppearances") || false),
```

Comment:
Given that this should be a boolean value, can we please check *explicitly* for that to prevent any future surprises/bugs in corrupt documents?
For example, what if the entry happens to be a non-empty string or any other truthy value? 
```suggestion
        !collectFields && acroFormDict.get("NeedAppearances") === true,
```